### PR TITLE
Add a InitialSpawnDelay to CrateSpawner

### LINKED ISSUE #9055: Delay for CrateSpawner
I understand that the unit duplication crate has been removed from all mods (which is needed). However I think this is just a bandaid. 

In the first 5 minutes of any mod, 1v1 especially, are critical. Crates really mess up games in these first 5 minutes. Unit dupe crates are the worst, then light tank crates in RA (light tank husk blocking REF or WF) then money crates. Even death crates are annoying as early map control and LoS is paramount. 

This is especially true on very small 1v1 maps.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25724881-delay-for-cratespawner?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes #9055.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/CrateSpawner.cs [hunk 1]
```diff
@@ -18,22 +18,25 @@ namespace OpenRA.Mods.Common.Traits
...
-		[Desc("Maximum number of crates")]
+		[Desc("Maximum number of crates.")]
 		public readonly int Maximum = 255;
 
-		[Desc("Average time (seconds) between crate spawn")]
+		[Desc("Average time (seconds) between crate spawn.")]
 		public readonly int SpawnInterval = 180;
 
+		[Desc("Delay (in seconds) before the first crate spawns.")]
+		public readonly int InitialSpawnDelay = 300;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/World/CrateSpawner.cs]: Can we have both of these in ticks to match our conventions in most other traits?

File: OpenRA.Mods.Common/Traits/World/CrateSpawner.cs [hunk 2]
```diff
@@ -18,22 +18,25 @@ namespace OpenRA.Mods.Common.Traits
...
+
 		[Desc("Which terrain types can we drop on?")]
 		public readonly string[] ValidGround = { "Clear", "Rough", "Road", "Ore", "Beach" };
 
 		[Desc("Which terrain types count as water?")]
 		public readonly string[] ValidWater = { "Water" };
 
-		[Desc("Chance of generating a water crate instead of a land crate")]
+		[Desc("Chance of generating a water crate instead of a land crate.")]
 		public readonly float WaterChance = .2f;
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/World/CrateSpawner.cs]: If you're cleaning up anyway, maybe change this to integer math?

**Comment by abcdefg30** [comment]
Updated.

**Comment by penev92** [comment]
Needs a rebase.

File: OpenRA.Mods.Common/Traits/World/CrateSpawner.cs [hunk 3]
```diff
@@ -88,9 +93,8 @@ public void Tick(Actor self)
 
 		void SpawnCrate(Actor self)
 		{
-			var threshold = 100;
-			var inWater = self.World.SharedRandom.NextFloat() < info.WaterChance;
-			var pp = ChooseDropCell(self, inWater, threshold);
+			var inWater = self.World.SharedRandom.Next() < info.WaterChance;
```
Comment (penev92) [inline OpenRA.Mods.Common/Traits/World/CrateSpawner.cs]: I believe you need to limit this to 100.
`self.World.SharedRandom.Next(100)`

**Comment by abcdefg30** [comment]
Updated and rebased.

Sorry that this rot that long.

**Comment by atlimit8** [comment]
Needs another rebase.

**Comment by abcdefg30** [comment]
Rebased.

**Comment by reaperrr** [comment]
Sorry, needs another rebase.

**Comment by reaperrr** [comment]
Looks good and works in-game, so :+1:

**Comment by abcdefg30** [comment]
Rebased again.

**Comment by obrakmann** [comment]
Works and looks fine. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/65a0173e42a95c0e3341ee1e4c712945b53c93db)

**Pull request merged**